packages/server/src/manifest-generator.ts

```diff
@@ -5,59 +5,86 @@ import { throwOnErrorEvent } from '@effection/events';
 import * as fs from 'fs';
 import * as globby from 'globby';
 import * as path from 'path';
+import { EslintValidator } from './validators/eslint-validator';
+import { Validator, OrchestratorState, BundlerState } from './orchestrator/state';
+import { Atom, Slice } from '@bigtest/atom';
 
 const { writeFile, mkdir } = fs.promises;
 
 interface ManifestGeneratorOptions {
   delegate: Mailbox;
   files: string[]; 
   destinationPath: string;
+  atom: Atom<OrchestratorState>;
 };
 
-function* writeManifest(options: ManifestGeneratorOptions) {
-  let files = yield globby(options.files);
+function* writeManifest({ bundlerSlice, destinationPath, ...options }: ManifestGeneratorOptions & { validator: Validator; bundlerSlice: Slice<BundlerState, OrchestratorState> }) {
+  bundlerSlice.update(() => ({ type: 'VALIDATING' }));
 
-  let manifest = 'let load = (res) => res.default || res;\n';
-  manifest += 'const children = [\n';
+  let files: string[] = yield globby(options.files);
 
-  for(let file of files) {
-    let filePath = "./" + path.relative(path.dirname(options.destinationPath), file);
-    manifest += `  Object.assign({}, load(require(${JSON.stringify(filePath)})), { path: ${JSON.stringify(file)} }),\n`;
-  }
+  let validState = options.validator.validate(options.files);
```

Comment:
Should validation be an `Operation` since it could be asynchronous? If not now, then in the future. As we add validators (typescript for example) then we can compose them into a single validation that runs in parallel with the others.

[1]: https://github.com/thefrontside/bigtest/issues/279

Reply:
now, this is a good question.  Does type checking lie here or in rollup.  My initial thoughts were rollup.  Writing your own typescript checker that is performant is a lot of code and it is pretty much solved by one of the many rollup plugins.  I think I want to try that approach first.  The rollup plugins do a lot of caching etc and I don't think we want to go down that road just yet.

I did not want to go overboard on validators for now as there is no requirement for multiple validators yet and I might be creating a structure for over-engineering for something that might never happen.

If there are extra validators then absolutely they should be operations and parallelised.

Reply:
having thought about it, I think you are right.  I've had a go.  Might be wrong to do this with a 1 element array.